pdc_client/__init__.py

```diff
@@ -161,7 +258,13 @@ def decode(req):
 
         content_type = "application/json"
         encode = json.dumps
-        self.client = beanbag.BeanBag(url, session=self.session, fmt=(content_type, encode, decode))
+        # In get_paged function it returns an interator, so user can get each pages quickly in plugins.
+        # The original call function of BeanBag returns the list.
+        # Keep the current codes and options in codes. Just init MultiPageBeanBag when is_multi is true.
+        if is_multi:
+            self.client = MultiPageBeanBag(url, session=self.session, fmt=(content_type, encode, decode))
```

Comment:
> The new class gets all pages but client needs to wait until all of them are fetched regardless of the page_size field value (with small page_size you would only make more request but client still needs to wait for all pages to be fetched). I think it could be better to use page_size=-1 so PDC server can send all results at once. But I still don't like that the page_size is ignored.

Yes, I think -1 also works for this case, but I think if we use page_size=-1, the data will still return after long time, because server should query all the data first. Maybe this is a little faster than than sending several requests.
But when the data is very big, maybe the http server has a time out and then return nothing, so I think return the whole data in one request maybe not a better way than the current changes, what's your thinking?